lib/tts_isolate.dart

```diff
@@ -47,6 +73,58 @@ void onStart(ServiceInstance service) async {
   WidgetsFlutterBinding.ensureInitialized();
   await Firebase.initializeApp();
 
+  final FlutterLocalNotificationsPlugin flutterLocalNotificationsPlugin =
+      FlutterLocalNotificationsPlugin();
+
+  await flutterLocalNotificationsPlugin.initialize(
+      const InitializationSettings(
+          android: AndroidInitializationSettings('notification_icon')),
+      onDidReceiveBackgroundNotificationResponse: (payload) async {
+    if (payload.actionId == "DISABLE_TTS") {
+      TextToSpeechPlugin.stopSpeaking();
+    }
+  });
+
+  if (service is AndroidServiceInstance) {
+    service.on('setAsForeground').listen((event) {
+      service.setAsForegroundService();
+    });
+
+    service.on('setAsBackground').listen((event) {
+      service.setAsBackgroundService();
+    });
+  }
+
+  service.on('stopService').listen((event) {
+    service.stopSelf();
+  });
+
+  //bring to foreground
+  Timer.periodic(const Duration(seconds: 1), (timer) async {
```

Comment:
This will burn a lot of battery, I think the call only needs to be made once? Did you take this code from somewhere else?

Reply:
I was playing around with others ways to handle this, why I thought this way would be better for android but if the cost is too high lemme try and change it to my other way.